liquibase-core/src/main/java/liquibase/statement/ExecutablePreparedStatementBase.java

```diff
@@ -122,50 +102,21 @@ public void execute(PreparedStatementFactory factory) throws DatabaseException {
         }
     }
 
-    protected PreparedStatement getCachedStatement(String sql) {
-        if (lastPreparedStatement == null || lastPreparedStatementSql == null) {
-            return null;
-        }
-
-        boolean statementIsValid = true;
-        if (lastPreparedStatementSql.equals(sql)) {
-            try {
-                if (lastPreparedStatement.isClosed()) {
-                    statementIsValid = false;
-                }
-                if (statementIsValid) {
-                    final Connection connection = lastPreparedStatement.getConnection();
-                    if (connection == null || connection.isClosed()) {
-                        statementIsValid = false;
-                    }
-                }
-            } catch (SQLException e) {
-                statementIsValid = false;
-            }
-
-        } else {
-            statementIsValid = false;
-        }
-
-        if (!statementIsValid) {
-            JdbcUtil.closeStatement(lastPreparedStatement);
-            lastPreparedStatement = null;
-            lastPreparedStatementSql = null;
-        }
-
-        return lastPreparedStatement;
-    }
-
     protected void executePreparedStatement(PreparedStatement stmt) throws SQLException {
         if (database instanceof PostgresDatabase) {
             //postgresql's default prepared statement setup is slow for normal liquibase usage. Calling with QUERY_ONESHOT seems faster, even when we keep re-calling the same prepared statement for many rows in loadData
             try {
-                if (executeWithFlagsMethod == null) {
-                    executeWithFlagsMethod = stmt.getClass().getMethod("executeWithFlags", int.class);
-                    executeWithFlagsMethod.setAccessible(true);
+                if (postgresExecuteWithFlagsMethod == null) {
```

Comment:
Same thing here, we have the same comparison outside and inside of the synchronized block.

Reply:
@ettavolt don't we need to also make the same change you made on StandardObjectChangeFilter here too by making `executeWithFlagsMethod` a volatile field as well?

Reply:
`postgresExecuteWithFlagsMethod` is already defined as `volatile`. (It was earlier version that used `executeWithFlagsMethod` name).

Reply:
You're totally right, I think I saw it wrong. Thank you!